src/mcmc.jl

```diff
@@ -1,13 +1,111 @@
-using ProgressMeter
 
 """
-    mcmc(gp::GPBase; kwargs...)
+    effective_sample_size(chain::AbstractMatrix)
 
-Runs Hamiltonian Monte Carlo algorithm for estimating the hyperparameters of Gaussian process `GPE` and the latent function in the case of `GPA`.
+Routine for computing the effective sample size as in,
+
+Gelman, Andrew, et al., 2013. Bayesian Data Analysis. 3rd.
+
+The samples are assuemd to be in column major order.
 """
-function mcmc(gp::GPBase; nIter::Int=1000, burn::Int=1, thin::Int=1, ε::Float64=0.1,
+
+function effective_sample_size(X::AbstractMatrix)
```

Comment:
Why does `compute_ess` need to be defined inside of `effective_sample_size`?

Reply:
The name is indeed confusing. I'll change it.